### PR TITLE
Add @Deprecated annotation support for proto file.

### PR DESCRIPTION
**Pull request by pherl**

### CODE REVIEW

File: src/google/protobuf/compiler/java/java_file.cc [hunk 1]
```diff
@@ -248,9 +248,12 @@ void FileGenerator::Generate(io::Printer* printer) {
   PrintGeneratedAnnotation(
       printer, '$', options_.annotate_code ? classname_ + ".java.pb.meta" : "");
   printer->Print(
-      "public final class $classname$ {\n"
+      "$deprecation$public final class $classname$ {\n"
```
Comment (xfxyjwf) [inline src/google/protobuf/compiler/java/java_file.cc]: Is there a test to check this annotation?

Comment (pherl) [reply src/google/protobuf/compiler/java/java_file.cc]: I only added a proto to make sure it compiles. Manually checked the annotations were generated though.

I can add a test to verify those annotations in a later PR (when I get time...)

File: java/core/src/test/proto/com/google/protobuf/deprecated_file.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+syntax = "proto3";
+
+package deprecated_file;
+
```
Comment (xfxyjwf) [inline java/core/src/test/proto/com/google/protobuf/deprecated_file.proto]: Can you add a TODO in this file (that we need a test to check the annotation)?

Comment (pherl) [reply java/core/src/test/proto/com/google/protobuf/deprecated_file.proto]: Done.

**Pull request merged**